internal/chatgpt/chatgpt.go

```
@@ -0,0 +1,70 @@
+package chatgpt
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/PullRequestInc/go-gpt3"
+	"github.com/slack-go/slack"
+	"github.com/xplorer-io/xplorers-bot-go/internal/config"
+	islack "github.com/xplorer-io/xplorers-bot-go/internal/slack"
+	"github.com/xplorer-io/xplorers-bot-go/internal/ssm"
+)
+
+type ChatGpt struct {
+	client gpt3.Client
+}
+
+func GetChatGPTApi(env *config.Config) (*ChatGpt, error) {
+	chatgptApiKey, GetSsmParameterErr := ssm.GetSsmParameter(env.ChatgptApiKeySsmPath)
+	if GetSsmParameterErr != nil {
+		return nil, GetSsmParameterErr
+	}
+
+	return &ChatGpt{
+		client: gpt3.NewClient(chatgptApiKey),
+	}, nil
+}
+
+func (c *ChatGpt) AskXplorersBot(slackEvent url.Values, env *config.Config) (*slack.Blocks, error) {
+	chatGptResponse, AskChatGptErr := c.AskChatGpt(slackEvent.Get("text"), env)
+	if AskChatGptErr != nil {
+		return nil, AskChatGptErr
+	}
+
+	slackResponse := GetFormattedChatGptResponseForSlack(chatGptResponse)
```

All function other than main should return an error

```
	slackResponse, err := GetFormattedChatGptResponseForSlack(chatGptResponse)
	if err != nil {
		return nil, fmt.Errorf("could not format response for slack: %v", err)
	}
```